PPO/util.py

```diff
@@ -25,31 +26,87 @@ def train(config):
       action, log_prob = agent.act(torch.FloatTensor(state))
       next_state, reward, done, _ = env.step(action.item())
 
-      agent.mem.add(torch.FloatTensor(state), action, reward, log_prob, done)
+      agent.add_to_mem(state, action, reward, log_prob, done)
 
       # Update 
       state = next_state
       score += reward
 
-      # Book Keeping
-      scores_deque.append(score)
-      scores.append(score)
-
       if steps >= config.update_every:
         agent.learn(config.num_learn)
         agent.mem.clear()
         steps = 0
 
       if done:
         break 
+
+    # Book Keeping
+    scores_deque.append(score)
+    scores.append(score)
+    average_scores.append(np.mean(scores_deque))
+    
       
     if i_episode % 10 == 0:
       print("\rEpisode {}	Average Score: {:.2f}	Score: {:.2f}".format(i_episode, np.mean(scores_deque), score), end="")
     if i_episode % 100 == 0:
       print("\rEpisode {}	Average Score: {:.2f}".format(i_episode, np.mean(scores_deque)))   
     
     if np.mean(scores_deque) > config.win_condition:
-      print("\rEnvironment Solved!")
+      print("\nEnvironment Solved!")
       break
 
-  return scores
+  return scores, average_scores
+
+def train_pixel(config):
```

Comment:
Should probably move this to somewhere else and not util, seems like a heavy util function. It's more like a main.

What I usually do is an Experiment class that encapsulates the current config and provides a train and evaluation function. Something we could consider.